### DISCUSSION TITLE
displacy.js for NER

### DISCUSSION THREAD

**Discussion by ch-sander** [explosion/spaCy#13239] [Help: Best practices]
https://github.com/explosion/spaCy/blob/e2a3952de51abb2620b4ff799ac461c87fec7bb4/website/docs/usage/visualizers.mdx#L417

I might fail to see where but [displacy.js](https://github.com/explosion/displacy/blob/master/assets/js/displacy.js) doesn't seem to render the NER tags but only dependency arcs.

**Comment by svlandeg** [comment]
Hi!

Can you explain a bit more what you're trying to do and what you're looking for? The repo https://github.com/explosion/displacy has been archived since 2018, as the displaCy visualizers have been integrated into the core library since spaCy v2.0.

**Comment by svlandeg** [comment]
You can find the docs on how to use Displacy to visualise NER here: https://spacy.io/usage/visualizers#ent

**Reply by ch-sander** [reply]
Thank you! I looked at the docs first and found this [caveat](https://spacy.io/usage/visualizers#webapp):

> If you want to use the visualizers as part of a web application, for example to create something like our [online demo](https://explosion.ai/demos/displacy), it’s not recommended to only wrap and serve the displaCy renderer. Instead, you should only rely on the server to perform spaCy’s processing capabilities, and use a client-side implementation like [displaCy.js](https://github.com/explosion/displacy) to render the JSON-formatted output.

Hence, I looked into `displacy.js`

Usecase: I render a HTML NER div within a closed database (Ninox). I currently call my own API (flask/python/spacy) but thought I might make use of a JS to directly render my JSON into the HTML markup.

**Reply by svlandeg** [reply]
Right! You can also have a look at the (old and archived) repo https://github.com/explosion/displacy-ent for inspiration!
